From f6fd6ad123fecfd59402664ce4c0e7413064c407 Mon Sep 17 00:00:00 2001
From: Chris Zhong <zyw@rock-chips.com>
Date: Wed, 8 Mar 2017 10:27:57 +0800
Subject: [PATCH] UPSTREAM: drm/rockchip: cdn-dp: add more log for video config

In order to analyze some video config failed, add some useful
printouts.

Change-Id: Icd16bb1d065b33163b6f0ce0d99c2a63d77cc915
Signed-off-by: Chris Zhong <zyw@rock-chips.com>
Signed-off-by: Sean Paul <seanpaul@chromium.org>
Link: http://patchwork.freedesktop.org/patch/msgid/1488940077-22297-4-git-send-email-zyw@rock-chips.com
(cherry picked from commit 9346ab7d72c0e0526bd9cea70fa4ad768fe27c20)
---
 drivers/gpu/drm/rockchip/cdn-dp-reg.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/cdn-dp-reg.c b/drivers/gpu/drm/rockchip/cdn-dp-reg.c
index b8e1a0f9754d..32c725ca9976 100644
--- a/drivers/gpu/drm/rockchip/cdn-dp-reg.c
+++ b/drivers/gpu/drm/rockchip/cdn-dp-reg.c
@@ -671,6 +671,10 @@ int cdn_dp_config_video(struct cdn_dp_device *dp)
 		rem = do_div(symbol, 1000);
 		if (tu_size_reg > 64) {
 			ret = -EINVAL;
+			DRM_DEV_ERROR(dp->dev,
+				      "tu error, clk:%d, lanes:%d, rate:%d\n",
+				      mode->clock, dp->link.num_lanes,
+				      link_rate);
 			goto err_config_video;
 		}
 	} while ((symbol <= 1) || (tu_size_reg - symbol < 4) ||
-- 
2.35.3

